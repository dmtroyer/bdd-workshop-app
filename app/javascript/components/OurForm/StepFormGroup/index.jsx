import React from 'react';
import { FormGroup, Input, Label } from 'reactstrap';
import PropTypes from 'prop-types';

const StepFormGroup = ({
  step,
  label,
  type,
  placeholder,
  values,
  handleChange,
}) => (
  <FormGroup>
    <Label for={step}>{label}</Label>
    <Input
      type={type || 'text'}
      name={step}
      value={values[step]}
      onChange={handleChange}
      placeholder={placeholder}
    />
  </FormGroup>
);
StepFormGroup.propTypes = {
  step: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  values: PropTypes.object,
  handleChange: PropTypes.func,
};

export default StepFormGroup;
