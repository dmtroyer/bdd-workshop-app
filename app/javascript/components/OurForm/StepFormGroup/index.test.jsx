import React from 'react';
import { shallow } from 'enzyme';
import StepFormGroup from '.';

describe('StepFormGroup', () => {
  it('Renders a register form', () => {
    const wrapper = shallow(
      <StepFormGroup
        values={{}}
        step="email"
        label="Email label"
        type="email"
        placeholder="placeholder text"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
