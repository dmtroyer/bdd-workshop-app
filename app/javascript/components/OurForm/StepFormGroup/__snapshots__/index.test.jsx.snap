// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepFormGroup Renders a register form 1`] = `
<FormGroup
  tag="div"
>
  <Label
    for="email"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Email label
  </Label>
  <Input
    name="email"
    placeholder="placeholder text"
    type="email"
  />
</FormGroup>
`;
